File: widgetssdk/src/main/java/com/glia/widgets/fileupload/domain/GetFileAttachmentsUseCase.java
package com.glia.widgets.fileupload.domain;

import com.glia.widgets.fileupload.FileAttachmentRepository;
import com.glia.widgets.fileupload.model.FileAttachment;

import java.util.List;

public class GetFileAttachmentsUseCase {
    private final FileAttachmentRepository repository;

    public GetFileAttachmentsUseCase(FileAttachmentRepository repository) {
        this.repository = repository;
    }

    public List<FileAttachment> execute() {
        return repository.getFileAttachments();
    }
}

File: widgetssdk/src/main/java/com/glia/widgets/internal/fileupload/domain/GetFileAttachmentsUseCase.kt
package com.glia.widgets.internal.fileupload.domain

import com.glia.widgets.internal.fileupload.FileAttachmentRepository
import com.glia.widgets.internal.fileupload.model.LocalAttachment

internal class GetFileAttachmentsUseCase(private val repository: FileAttachmentRepository) {
    operator fun invoke(): List<LocalAttachment> {
        return repository.getFileAttachments()
    }
}
